pkgs/tools/misc/nncp/default.nix

```
@@ -0,0 +1,46 @@
+{ stdenv, go, fetchurl }:
+
+stdenv.mkDerivation rec {
+  pname = "nncp";
+  version = "5.3.3";
+
+  buildInputs = [ go ];
+
+  preConfigure = ''
+    export GOCACHE=$PWD/.cache
+  '';
+
+  makeFlags = [
+    "PREFIX=$$out"
+    "CFGPATH=/etc/nncp.hjson"
+    "SENDMAIL=/run/wrappers/bin/sendmail"
```

Is this a must requirement for runtime as well? If so, perhaps you should use:

```suggestion
    "SENDMAIL=${sendmail}/bin/sendmail"
```

While importing it properly and put it in the `buildInputs`.

---

It's optional, so I wanted just a more sensible path if someone has an MTA installed and wants to use mail.

---

I can see that the `sendmail` executable path is embedded in the resulting executable, according to this: https://github.com/jgoerzen/nncp/blob/46be2c9c706cd824fe78404bace49c88ec236bdd/Makefile#L21

Is this path configurable somehow from the program? If not, I'd test the program (with whatever is related to this sendmail feature) while using Nix's `sendmail`. Try it with a non absolute path as well.

---

$SENDMAIL is just used to set the default path in the generated config file. NNCP default is `/usr/sbin/sendmail`, so `/run/wrappers/bin/sendmail`, which is just a wrapper for the system MTA (if installed) in NixOS, seems to be the best choice here.

---

> $SENDMAIL is just used to set the default path in the generated config file. NNCP default is /usr/sbin/sendmail, so /run/wrappers/bin/sendmail, which is just a wrapper for the system MTA (if installed) in NixOS, seems to be the best choice here.

I see. You are right that if it's an optional feature that's using this workaround, we shouldn't constraint users of this package to have a derivation of `sendmail` in their `/nix/store/`. But you should be aware that you are choosing a very specific Nix-like path to be embedded in the executable. If you are absolutely sure that this is configurable, you can mark this conversation as resolved.

---

would you be more happy with `/usr/bin/env sendmail` instead?

---

I has to be just a path w/o arg to make sense. I'll leave it in /run/wrappers, it is only referenced in the default-config-file generator.